cpp/src/io/csv/reader_impl.cu

```diff
@@ -349,7 +349,12 @@ table_with_metadata reader::impl::read(size_t range_offset,
       if (column_types[active_col].id() == type_id::EMPTY) {
         column_types[active_col] = data_type{STRING};
       }
-      out_buffers.emplace_back(column_types[active_col], num_records, true, stream, mr_);
+      const bool is_final_allocation = column_types[active_col].id() != type_id::STRING;
```

Comment:
This smells like a WAR as opposed to a real fix. Is there a more permanent fix for this?

Reply:
There are two reasons why the condition exists here:
1. `make_column` does not move the input buffer for string columns, but does so for every other type.
2. We have an extra step to remove double quotes, after the data is decoded.

The second reason can be eliminated by removing the double quotes as the data is decoded, instead of a separate step (should also be faster).
I don't know if or how we can make the `make_column` behavior consistent for all types. Thoughts?

Without changes to `make_column`, to me it looks like improving this code requires refactoring of the function/reader (which is expected at a later point). Let me know what you think.

Reply:
> make_column does not move the input buffer for string columns, but does so for every other type.

This seems like a low-hanging fruit to remedy (not in this PR).

Reply:
So do we agree to merge this PR and move on to the next PR for `make_column`?

Reply:
Yep. Just wanted to know what the longer term fix is.